delivery/app.py

```diff
@@ -6,6 +6,44 @@
 from delivery.error_handling import CustomError, WrongIdError
 from flasgger import Swagger
 from marshmallow import Schema, fields, ValidationError
+from abc import ABC, abstractmethod
+
+
+class AbstractRestaurantRepository(ABC):
+    @abstractmethod
+    def save(self, content, restaurant_id):
+        pass
+
+    @abstractmethod
+    def get_all(self):
+        pass
+
+    @abstractmethod
+    def get_by_id(self, id):
+        pass
+
+
+class MemoryRestaurantRepository(AbstractRestaurantRepository):
+    def __init__(self, path):
```

Comment:
I would try to keep the file reading code away from the constructor:
1. keep constructor simple, like:
  ```python
  def __init__(self, restaurants=None ):
    self.restaurants = restaurants or []
  ```
2. implement a separate function to read the list from the file, e.g. `read_restaurants()`, and `repository = MemoryRestaurantRepository(read_restaurants())`

Reply:
fixed